src/helpers/figures/charts/chart_common.ts

```diff
@@ -428,3 +437,118 @@ export function getDefinedAxis(definition: ChartWithAxisDefinition): {
   useLeftAxis ||= !useRightAxis;
   return { useLeftAxis, useRightAxis };
 }
+
+export function getTrendDatasetForBarChart(config: TrendConfiguration, dataset: any) {
+  const values: number[] = [];
+  const labels: number[] = [];
+  for (let i = 0; i < dataset.data.length; i++) {
+    values.push(dataset.data[i]);
+    labels.push(i + 1);
+  }
+
+  const datasetLength = dataset.data.length;
+  const newLabels: number[] = [];
+  for (let i = 0.5; i < datasetLength + 0.5; i += 0.1) {
+    newLabels.push(i);
+  }
+
+  const newValues = interpolateData(config, values, labels, newLabels);
+  if (!newValues.length) {
+    return;
+  }
+  return {
+    ...dataset,
+    type: "line",
+    xAxisID: "x1",
+    label: _t("Trend line for %s", dataset.label ?? ""),
+    data: newValues,
+    order: -1,
+    pointRadius: 0,
+    backgroundColor: config.color ?? setColorAlpha(dataset.backgroundColor, 0.5),
+    borderColor: config.color ?? setColorAlpha(dataset.borderColor, 0.5),
+    borderDash: [5, 5],
+  };
+}
+
+export function getTrendDatasetForLineChart(
+  config: TrendConfiguration,
+  dataset: any,
+  labelsAsText?: boolean
+) {
+  const values: number[] = [];
+  const labels: number[] = [];
+  const datasetLength = dataset.data.length;
+  const newLabels: number[] = [];
+  if (labelsAsText) {
```

Comment:
not the correct condition, labelAsText can be false whithout the chart having `{x, y}` points. You should test the chart axis type.

The trenline is breaking if there is non-number strings in the labels